src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingChannelExistenceChecker.java

```diff
@@ -0,0 +1,293 @@
+import net.snowflake.ingest.connection.OAuthCredential;
+import net.snowflake.ingest.connection.RequestBuilder;
+import net.snowflake.ingest.connection.ServiceResponseHandler;
+import net.snowflake.ingest.streaming.SnowflakeStreamingIngestChannel;
+import net.snowflake.ingest.utils.BackOffException;
+import net.snowflake.ingest.utils.Constants;
+import net.snowflake.ingest.utils.ErrorCode;
+import net.snowflake.ingest.utils.HttpUtil;
+import net.snowflake.ingest.utils.SFException;
+import net.snowflake.ingest.utils.SnowflakeURL;
+
+/**
+ * Class which handles REST API call directly for a Streaming channel and obtains a response of its
+ * existence.
+ *
+ * <p>This is a bit different from what we are doing in {@link
+ * SnowflakeStreamingIngestChannel#getLatestCommittedOffsetToken()} wherein we first have to Open A
+ * Channel and then make another call for getOffsetToken.
+ *
+ * <p>A method (checkChannelExistence) in this class bypasses that open Channel call and directly
+ * creates the payload for Get Status {@link
+ * StreamingChannelExistenceChecker#CHANNEL_STATUS_ENDPOINT} API.
+ */
+public class StreamingChannelExistenceChecker {
+  private static final KCLogger LOGGER =
+      new KCLogger(StreamingChannelExistenceChecker.class.getName());
+
+  private static final String DEBUG_PREFIX = "Channel Existence Checker - Kafka Connector";
+
+  private static final String CHANNEL_STATUS_ENDPOINT = "/v1/streaming/channels/status/";
+
+  // This is just a placeholder String, it has no value since we are not making a call through a
+  // Client.
+  private static final String STREAMING_CUSTOM_REST_API_CLIENT = "KC_CHANNEL_EXISTENCE_CHECKER";
+  private final String tableName;
+  private final Map<String, String> connectorConfig;
+
+  // Using SnowflakeURL from IngestSDK
+  private final SnowflakeURL snowflakeURL;
+
+  // We always set clientSequencer to 0.
+  private final Long clientSequencer;
+
+  @VisibleForTesting
+  StreamingChannelExistenceChecker(
+      String tableName, Map<String, String> connectorConfig, final Long clientSequencer) {
+    this.tableName = tableName;
+    this.connectorConfig = connectorConfig;
+    this.snowflakeURL = new SnowflakeURL(this.connectorConfig.get(Utils.SF_URL));
+    this.clientSequencer = clientSequencer;
+  }
+
+  /**
+   * C'tor for StreamingChannelExistenceChecker
+   *
+   * @param tableName TableName for checking Channel Name in it.
+   * @param connectorConfig connectorConfig passed in SF Kafka Connect Configuration
+   */
+  public StreamingChannelExistenceChecker(String tableName, Map<String, String> connectorConfig) {
+    this(tableName, connectorConfig, 0L);
+  }
+
+  /**
+   * Determines if a Channel Exists or not for a table defined in this class's ctor.
+   *
+   * @param channelName Name of Channel to Check for Existence
+   * @return True if it exists, false if it doesn't.
+   */
+  public boolean checkChannelExistence(final String channelName) {
```

Comment:
nit: checkChannelExists so that it's easier to intuit that returning TRUE means the channel exists

Reply:
makes sense, will do!